### PR TITLE
Add object value class generation

### PR DESCRIPTION
**Pull request by SupunS**

### CODE REVIEW

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_constants.bal [hunk 1]
```diff
@@ -206,4 +207,4 @@ const string MAP_TYPE = "org/ballerinalang/jvm/types/BMapType";
 const string UNION_TYPE = "org/ballerinalang/jvm/types/BUnionType";
 const string RECORD_TYPE = "org/ballerinalang/jvm/types/BRecordType";
 const string OBJECT_TYPE = "org/ballerinalang/jvm/types/BObjectType";
-const string BFIELD = "org/ballerinalang/jvm/types/BField";
+const string BFIELD = "org/ballerinalang/jvm/types/BField";
```
Comment (vinok88) [inline compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_constants.bal]: Add newline

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_constants.bal [hunk 2]
```diff
@@ -191,9 +191,10 @@ final int V1_8 = 0 << 16 | 52;
 // jvm runtime values related classes
 const string MAP_VALUE = "org/ballerinalang/jvm/values/MapValue";
 const string ARRAY_VALUE = "org/ballerinalang/jvm/values/ArrayValue";
+const string OBJECT_VALUE = "org/ballerinalang/jvm/values/ObjectValue";
```
Comment (Kishanthan) [inline compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_constants.bal]: Shall we use some consistence naming convention for the constants here? May be we an use one pattern for jvm classes and one or ballerina types, values ?

Comment (SupunS) [reply compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_constants.bal]: The current convention is as follows:

- Values: \<typename\>_TYPE 
- Type names: \<typename\>_VALUE 
- Other JVM classes: same name as their java class name

There are some cases that belongs to two of these categories. eg `java.lang.Object`. Those were named according to the 3rd category.

Comment (Kishanthan) [reply compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_constants.bal]: ack, if we have a identifiable pattern

**Pull request merged**